Tribler/Core/Modules/watch_folder.py

```diff
@@ -0,0 +1,51 @@
+import hashlib
+import logging
+import os
+from twisted.internet.task import LoopingCall
+from Tribler.Core.TorrentDef import TorrentDef
+from Tribler.Core.Utilities.utilities import fix_torrent
+from Tribler.Core.simpledefs import NTFY_CORRUPT_TORRENT, NTFY_INSERT
+from Tribler.dispersy.taskmanager import TaskManager
+
+
+WATCH_FOLDER_CHECK_INTERVAL = 10
+
+
+class WatchFolder(TaskManager):
+
+    def __init__(self, session):
+        super(WatchFolder, self).__init__()
+
+        self._logger = logging.getLogger(self.__class__.__name__)
+        self.session = session
+        self.watch_folder_check_task = None
+
+    def start(self):
+        self.watch_folder_check_task = self.register_task("check watch folder", LoopingCall(self.check_watch_folder))
```

Comment:
You don't need to keep a reference to this task, just register_task(foo).start(bar) and keep it simple